scale/job/models.py

```diff
@@ -1309,10 +1307,29 @@ def configure_docker_params(self, framework_id, workspaces):
             configuration.add_job_task_workspace(workspace_name, MODE_RW)
 
         # Configure any Docker parameters needed for workspaces
-        configuration.configure_workspace_docker_params(framework_id, self.id, workspaces)
+        configuration.configure_workspace_docker_params(self, workspaces, docker_volumes)
 
         self.configuration = configuration.get_dict()
 
+    def get_cluster_id(self):
+        """Gets the cluster ID for the job execution. This call is only valid after the job execution has been scheduled
+        (is no longer queued).
+
+        :returns: The cluster ID for the job execution
+        :rtype: string
+
+        :raises Exception: If the job execution is still queued
+        """
+
+        if self.status == 'QUEUED':
+            raise Exception('cluster_id is not set until the job execution is scheduled')
```

Comment:
I made a custom exception for critical Scale logic bugs and used it in all of the places you mentioned.